repository/gorm/message.go

```diff
@@ -294,35 +296,114 @@ func (repo *Repository) SetMessageUnread(userID, messageID uuid.UUID, noticeable
 	if userID == uuid.Nil || messageID == uuid.Nil {
 		return repository.ErrNilID
 	}
+	s := set.UUID{}
+	if noticeable {
+		s.Add(userID)
+	}
+	return repo.BulkSetMessageUnread([]uuid.UUID{userID}, messageID, s)
+}
+
+// BulkSetMessageUnread implements MessageRepository interface.
+func (repo *Repository) BulkSetMessageUnread(userIDs []uuid.UUID, messageID uuid.UUID, noticeable set.UUID) error {
+	if userIDs == nil || messageID == uuid.Nil {
```

Comment:
スライスが`nil`であるときに`repository.ErrNilID`を返すのは適切でないと思います。`userIDs == nil`の場合はこの次の`len`のチェックで弾かれるのでここでは見なくて良いんじゃないでしょうか
```suggestion
	if messageID == uuid.Nil {
```